### PR TITLE
Adding first usage of google apis and extending code arquitecture for extensibility

### LINKED ISSUE #16: Complete GoogleSheetRepository to read orgchart
In order to unblock first end-to-end executions we should implement a read method for an orgchart sheet file that will be place in google drive

### LINKED ISSUE #20: Googlea APIs: set correct scopes
we have

```
SCOPES = [
        'https://www.googleapis.com/auth/drive',
        'https://www.googleapis.com/auth/script.projects',
        'https://www.googleapis.com/auth/gmail.readonly',
        'https://www.googleapis.com/auth/documents.readonly',
        'https://www.googleapis.com/auth/spreadsheets.readonly'
    ]
```

And we should review what scopes do we really need

### LINKED ISSUE #21: DI module is always dev. We should parametrized it
We should inject production module for production code and not the dev one

### PR DESCRIPTION
**Pull request by eduardogr**
- Changing documentation structure
    - moving to doc/
- Set correct scopes for google APIs, fix #20 
- Updating models
    - new Employee model to represent just an Employee.
    - new Employee model, old Employee now is EmployeeNode
- Add storages.py: concrete implementations for core.DataRepository
- Add communications_channels.py: concrete implementations for core.ComunicationsProvider
- Adding a little setup code just to play (it's more like a reminder thing, i have to fix it)

fix #21, adding parametrization of DI.Module
fix #16, code to read google sheets within storages.GoogleStorage

### CODE REVIEW

**Comment by eduardogr** [comment]
After this merge 2 more issues could be done
- https://github.com/eduardogr/evalytics/issues/22
- https://github.com/eduardogr/evalytics/issues/24

File: evalytics/server/adapters.py [hunk 1]
```diff
@@ -11,20 +11,29 @@ def create_initial_eval_suite(
         """
         eval_suite = EvalSuite()
 
-        for employee in org_chart:
```
Comment (manutero) [inline evalytics/server/adapters.py:14]: sad

File: evalytics/server/communications_channels.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+from googleapiclient.discovery import build
```
Comment (manutero) [inline evalytics/server/communications_channels.py:1]: too many `@classmethod` for my body.  I'd use regular `self` methods instead to be honest. 

```python
gmail_channel.send('peter.parker@gmail.com', data)
```

sounds good IMO
```

File: evalytics/server/core.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+
+
```
Comment (manutero) [inline evalytics/server/core.py:2]: to be honest, I start doubting about this "select which implementation" castle we're building here. I start thinking that it's not needed actually. 

#trolling If we change some names adding some fancy words like `Provider` `ManagerProvider` ... we could turn this into a Java app

File: evalytics/server/models.py [hunk 4]
```diff
@@ -6,7 +6,32 @@
...
 @dataclass
-class Employee(NodeMixin):
+class Team:
+
+    def __init__(self, name: str,
+                 manager: str = None,
+                 manager_one_level_up: str = None):
+        self.__name = name
+        self.__manager = manager
+        self.__manager_one_level_up = manager_one_level_up
```
Comment (manutero) [inline evalytics/server/models.py:16]: so this is "who reach first the class, defines the naming convention", just asking

Comment (eduardogr) [reply evalytics/server/models.py:16]: hahaha nope, we should end having just one

**Pull request merged**